src/collections/utils.ts

```diff
@@ -75,18 +74,17 @@ function getBaseAPIPath(pathFromUrl?: string | null) {
  * @returns URL as string
  */
 export function createAstraUri (
-    databaseId: string,
-    region: string,
-    keyspace: string,
+    apiEndPoint: string,
+    keyspace?: string,
     applicationToken?: string,
     baseApiPath?: string,
     logLevel?: string,
     authHeaderName?: string,
 ) {
-    const uri = new url.URL(`https://${databaseId}-${region}.apps.astra.datastax.com`);
+    const uri = new url.URL(apiEndPoint);
     let contextPath = '';
     contextPath += baseApiPath ? `/${baseApiPath}` : '/api/json/v1';
-    contextPath += `/${keyspace}`;
+    contextPath += `/${keyspace || 'default_keyspace'}`;
```

Comment:
I think it is more correct to do `keyspace ?? 'default_keyspace'` so passing an empty string to keyspace doesn't end up as 'default_keyspace'.

Reply:
solved